include/deal.II/base/index_set.h

```diff
@@ -59,6 +59,145 @@ DEAL_II_NAMESPACE_OPEN
 class IndexSet
 {
 public:
+
+  typedef types::global_dof_index index_type;
+
+  class ElementIterator;
+  class IntervalIterator;
+
+  class IntervalAccessor
+  {
+  public:
+    IntervalAccessor(const IndexSet &idxset, index_type range_idx);
+    index_type n_elements() const;
+    bool is_valid() const;
+    ElementIterator begin() const;
+    ElementIterator end() const;
+  private:
+    bool operator == (const IntervalAccessor &other) const;
+    bool operator < (const IntervalAccessor &other) const;
```

Comment:
Don't you need to make these two functions public if you want people to use the class as an iterator?